tutorials/blink_led.rst

```diff
@@ -0,0 +1,378 @@
+
+.. image:: assets/blink_led_hardware.jpg
+   :alt: Hardware Requirements
+
+- A Raspberry Pi Pico or a an ESP32 microcontroller board with GPIO pins
+- A USB cable to connect the board to a laptop (macOS or Linux)
+- A LED preferrably red
+- A resistor (150 Ω or higher)
+- Few dupont wires
+- A breadboard
+
+You may also want to use a multimeter.
+
+If your multimeter has a diode testing feature (usually with the diode symbol in the Ohm-meter zone), you may want to check the LED still works by connecting the longer leg (anode) to the V/Ω input of the multimeter and the shorter leg (cathode) of the LED to the GND input.
+
+You may also want to check the resistance of the resistor you will use. If you have the specification of the LED you want to use, you can apply Ohm law to compute the proper resistor value. Usual boards often have a 5V pin but will supply a 3.3V level with GPIOs, so for a 10 mA 1.8V red LED, 150 Ω would work.
+
+Software Requirements
+---------------------
+
+This tutorial supposes you already installed AtomVM on your board and you have installed the appropriate build tools (rebar3, mix or gleam) for your favorite language on your desktop machine. For gleam, you will need to install packbeam and uf2tool (for Pico) or esptool.py (for ESP32).
+
+You will also need `minicom` or an alternate serial console software.
+
+Circuit
+-------
+
+Connect the LED to a GPIO pin (or any available GPIO pin) through a resistor to ground.
+
+.. figure:: build/assets/led_circuit.svg
+   :alt: LED blinking circuit diagram
+   :align: center
+   :width: 400px
+
+   Circuit diagram showing GPIO connected through a 150Ω resistor to the anode of a LED, with the LED cathode connected to ground.
+
+.. figure:: build/assets/led_breadboard.svg
+   :alt: LED blinking breadboard circuit diagram
+   :align: center
+   :width: 600px
+
+   Breadboard circuit diagram showing the physical connections for blinking an LED with a Raspberry Pi Pico.
+
+Creating and configuring the project
+------------------------------------
+
+Use build tools to generate an empty project.
+
+.. tab:: Erlang
+
+  Create a new directory and initialize a rebar3 project:
+
+  .. code-block:: bash
+
+     rebar3 new lib blink_led
+     cd blink_led
+
+  Your project structure should look like:
+
+  .. code-block:: text
+
+     blink_led/
+     ├── rebar.config
+     ├── src/
+     │   ├── blink_led.app.src
+     │   └── blink_led.erl
+
+  Add the following line to `rebar.config`:
+
+  .. code-block:: erlang
+
+     {plugins, [atomvm_rebar3_plugin]}.
+
+.. tab:: Elixir
+
+  Create a new directory and initialize a mix project:
+
+  .. code-block:: bash
+
+     mix new blink_led
+
+  Your project structure should look like:
+
+  .. code-block:: text
+
+     blink_led/
+     ├── mix.exs
+     ├── lib/
+     │   └── blink_led.ex
+     ├── test/
+     │   ├── blink_led_test.exs
+     │   └── test_helper.exs
+
+  Update the ``mix.exs`` file with the atomvm elements definition of the
```

Comment:
I'm not super savvy with restructuredtext, what is the significance of the double backticks here?